lib/linguist/heuristics.rb

```diff
@@ -75,6 +75,11 @@ def call(data)
     end
 
     # Common heuristics
+    CRegex = Regexp.union(
+        /^(char|double|short|int|long|float|unsigned|void)\w+\s*\(\s*void\s*\)/,
+        /\([\s\w\*]+restrict\s+/,
+        /^(char|short|int|long|float|unsigned)\s+\w+\s*\[\]\s*;/
```

Comment:
@pchaigno , yes, for the first and the third, the word boundary is more helpful as it will also match common c-types like `unsigned char`, `unsigned int`, `long long` or `unsigned long` made the changes